src/modules/75-cd/components/PipelineSteps/ServerlessInfraSpec/ServerlessInfraSpec.tsx

```diff
@@ -0,0 +1,316 @@
+import cx from 'classnames'
+import type { FormikProps } from 'formik'
+import { useVariablesExpression } from '@pipeline/components/PipelineStudio/PiplineHooks/useVariablesExpression'
+import { ConfigureOptions } from '@common/components/ConfigureOptions/ConfigureOptions'
+import { FormMultiTypeConnectorField } from '@connectors/components/ConnectorReferenceField/FormMultiTypeConnectorField'
+import { getIconByType } from '@connectors/pages/connectors/utils/ConnectorUtils'
+import { useStrings } from 'framework/strings'
+import type { GitQueryParams } from '@common/interfaces/RouteInterfaces'
+import { useQueryParams } from '@common/hooks'
+import { StageErrorContext } from '@pipeline/context/StageErrorContext'
+import { VariablesListTable } from '@pipeline/components/VariablesListTable/VariablesListTable'
+import { DeployTabs } from '@cd/components/PipelineStudio/DeployStageSetupShell/DeployStageSetupShellUtils'
+import type { StepViewType } from '@pipeline/components/AbstractSteps/Step'
+import type { ServerlessInfraTypes } from '@pipeline/utils/stageHelpers'
+import type { VariableMergeServiceResponse } from 'services/pipeline-ng'
+import type { ConnectorInfoDTO } from 'services/cd-ng'
+import { getValidationSchema, getValidationSchemaWithRegion } from '../PipelineStepsUtil'
+import stepCss from '@pipeline/components/PipelineSteps/Steps/Steps.module.scss'
+import css from './ServerlessInfraSpec.module.scss'
+
+export interface ServerlessSpecEditableProps {
+  initialValues: ServerlessInfraTypes
+  onUpdate?: (data: any) => void
+  stepViewType?: StepViewType
+  readonly?: boolean
+  template?: ServerlessInfraTypes
+  metadataMap: Required<VariableMergeServiceResponse>['metadataMap']
+  variablesData: ServerlessInfraTypes
+  allowableTypes: MultiTypeInputType[]
+  hasRegion?: boolean
+  formInfo: {
+    formName: string
+    type: ConnectorInfoDTO['type']
+    header: string
+    tooltipIds: {
+      connector: string
+      region?: string
+      stage: string
+    }
+  }
+}
+
+export const ServerlessSpecEditable: React.FC<ServerlessSpecEditableProps> = ({
+  initialValues,
+  onUpdate,
+  readonly,
+  allowableTypes,
+  hasRegion,
+  formInfo
+}): JSX.Element => {
+  const { accountId, projectIdentifier, orgIdentifier } = useParams<{
+    projectIdentifier: string
+    orgIdentifier: string
+    accountId: string
+  }>()
+  const { repoIdentifier, branch } = useQueryParams<GitQueryParams>()
+  const delayedOnUpdate = React.useRef(debounce(onUpdate || noop, 300)).current
+  const { expressions } = useVariablesExpression()
+  const { getString } = useStrings()
+
+  const { subscribeForm, unSubscribeForm } = React.useContext(StageErrorContext)
+
+  const formikRef = React.useRef<FormikProps<unknown> | null>(null)
+
+  React.useEffect(() => {
+    subscribeForm({ tab: DeployTabs.INFRASTRUCTURE, form: formikRef })
+    return () => unSubscribeForm({ tab: DeployTabs.INFRASTRUCTURE, form: formikRef })
+  }, [])
```

Comment:
Codacy has a fix for the issue: [React Hook React.useEffect has missing dependencies: 'subscribeForm' and 'unSubscribeForm'. Either include them or remove the dependency array.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9232297)
```suggestion
  }, [subscribeForm, unSubscribeForm])
```